localstack/services/s3/models.py

```diff
@@ -48,4 +49,40 @@ class S3Store(BaseStore):
     )
 
 
-s3_stores = AccountRegionBundle("s3", S3Store)
+class BucketCorsIndex:
+    def __init__(self):
+        self._cors_index_cache = None
+        self._bucket_index_cache = None
+
+    @property
+    def cors(self) -> Dict[str, CORSConfiguration]:
+        if self._cors_index_cache is None:
+            self._cors_index_cache = self._build_cors_index()
+        return self._cors_index_cache
+
+    @property
+    def buckets(self) -> Set[str]:
+        if self._bucket_index_cache is None:
+            self._bucket_index_cache = self._build_bucket_index()
+        return self._bucket_index_cache
+
+    def invalidate(self):
+        self._cors_index_cache = None
+        self._bucket_index_cache = None
+
+    @staticmethod
+    def _build_cors_index() -> Dict[BucketName, CORSConfiguration]:
+        result = {}
+        for account_id, regions in s3_stores.items():
+            result.update(regions[config.DEFAULT_REGION].bucket_cors)
```

Comment:
Is the index region specific?
Here the default region is used, while for the buckets it's the "global" region.

Reply:
Yes, it's because for the `S3Store` store, `bucket_cors` is defined as a cross-region attribute, which we can access from any region.
For moto backends, S3 data is stored in the `global` region. We do not have this in our stores. It's not region specific, we've done tests and the config is accessible from any region.